BedrockServer.cpp

```diff
@@ -879,12 +879,12 @@ void BedrockServer::worker(const SData& args,
                           << ((STimeNow() - preLockTime)/1000) << "ms.");
                 }
 
-                // If the command doesn't already have an httpsRequest from a previous peek attempt, try peeking it
-                // now. We don't duplicate peeks for commands that make https requests.
+                // If the command has any httpsRequests from a previous `peek`, we won't peek it again unless the
+                // command has specifically asked for that.
                 // If peek succeeds, then it's finished, and all we need to do is respond to the command at the bottom.
                 bool calledPeek = false;
                 bool peekResult = false;
-                if (!command.httpsRequests.size()) {
+                if (command.repeek || !command.httpsRequests.size()) {
```

Comment:
should we protect against a command forgetting to reset `repeek` to false by requiring that it have `repeek = true` AND has some non-complete https request?

Reply:
But we can't add the next https request until we're already in `peek`.

Reply:
We could reset `repeek` to false every time we peek, thus requiring the command to deliberately re-set it to `true` if it wants to peek again.

Reply:
im not sure that necessarily prevents what i'm looking to prevent, i think we should protect against a command getting stuck forever because repeek is set to true and never reset to false, thus meaning the command will loop peek forever until the server is stopped

Reply:
It's still subject to regular timeout rules.